Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx

```diff
@@ -0,0 +1,474 @@
+
+    const [ submitSolution ] = useSubmitContestSolutionMutation();
+    const [ getContestById ] = useLazyGetContestByIdQuery();
+    const [ getSubmissionsData ] = useLazyGetSubmissionResultsByProblemQuery();
+    const [ getContestUserParticipation ] = useLazyGetContestUserParticipationQuery();
+
+    const isCompete = participationType === 'compete';
+    const isModalOpen = Boolean(anchorEl);
+
+    const textColorClassName = getColorClassName(themeColors.textColor);
+    const lightBackgroundClassName = getColorClassName(themeColors.baseColor100);
+
+    const {
+        data,
+        isLoading,
+        error,
+    } = useGetContestRegisteredUserQuery({ id: contestId!.toString(), isOfficial: isCompete });
+
+    const { requirePassword, isOnlineExam, name, numberOfProblems, duration, id } = data || {};
+    const {
+        contest,
+        shouldEnterPassword,
+        participantsCount,
+        lastSubmissionTime,
+        userSubmissionsTimeLimit,
+        endDateTimeForParticipantOrContest,
+    } = userContestParticipationData || {};
+    const { problems, allowedSubmissionTypes = [] } = contest || {};
+    const {
+        memoryLimit,
+        timeLimit,
+        fileSizeLimit,
+        checkerName,
+    } = selectedContestDetailsProblem || {};
+
+    const fetchUserParticipationDetails = async () => {
+        setUserParticipationDataLoading(true);
+
+        try {
+            const { data: queryData } = await getContestUserParticipation({
+                id: contestId!,
+                isOfficial: isCompete,
+            });
+
+            setUserParticipationError('');
+            dispatch(setUserContestParticipationData({ participationData: queryData! }));
+            setUserParticipationDataLoading(false);
+        } catch {
+            setUserParticipationError('Error loading user participation data!');
+            dispatch(setUserContestParticipationData({ participationData: null }));
+            setUserParticipationDataLoading(false);
+        }
+    };
+
+    const onStrategyDropdownItemSelect = useCallback((s: any) => {
+        const submissionType = allowedSubmissionTypes.find((type: ISubmissionTypeType) => type.id === s.id);
+
+        setSelectedStrategyValue(s.id);
+        setSelectedSubmissionType(submissionType);
+    }, [ allowedSubmissionTypes ]);
+
+    const strategyDropdownItems = useMemo(
+        () => allowedSubmissionTypes?.map((item: any) => ({ id: item.id, name: item.name })),
+        [ allowedSubmissionTypes ],
+    );
+
+    useEffect(() => {
+        setSubmissionCode('');
+    }, [ selectedContestDetailsProblem ]);
+
+    // fetch contest data, when user has accepted online exam modal,
+    // entered password correctly and has accessed contest data
+    useEffect(() => {
+        if (!userContestParticipationData || userContestParticipationData.contest.id !== Number(contestId)) {
+            fetchUserParticipationDetails();
+        }
```

Comment:
Nice job including comments for such a complicated logic! I just don't understand how this check covers the modal and password logic.. Also why is this check - `userContestParticipationData.contest.id !== Number(contestId)`? Do we have a case when the `userContestParticipationData.contest.id` will be different from the current's `contestId`?

Reply:
no we do not have that case, this check covers that we do not make 2 reqeusts if we already have the same contest loaded in the slice. Basically for optimisation purposes is that check.